std/flags/mod.ts

```diff
@@ -51,82 +95,81 @@ function isNumber(x: unknown): boolean {
 
 function hasKey(obj: NestedMapping, keys: string[]): boolean {
   let o = obj;
-  keys.slice(0, -1).forEach(function(key: string): void {
-    o = (get(o, key) || {}) as NestedMapping;
+  keys.slice(0, -1).forEach(key => {
+    o = (get(o, key) ?? {}) as NestedMapping;
   });
 
   const key = keys[keys.length - 1];
   return key in o;
 }
 
+/** Take a set of command line arguments, with an optional set of options, and
+ * return an object representation of those argument.
+ *
+ *      const parsedArgs = parse(Deno.args);
+ */
 export function parse(
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  args: any[],
-  initialOptions: Partial<ArgParsingOptions> = {}
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-): { [key: string]: any } {
-  const options: ArgParsingOptions = {
+  args: string[],
+  options: Partial<ArgParsingOptions> = {}
```

Comment:
We should handle the options object / defaults by destructuring here, e.g. https://github.com/denoland/deno/blob/18076db8f004d70340a7a4317e0ac14c2ca56dee/std/fs/walk.ts#L61-L72